### PR TITLE
Fix/raycast filtering

### PR DESCRIPTION
**Pull request by jullianpalma**
right now the raycast is working pretty well (on my end at least) for stuff like the random boxes strewn about. but it BARELY works on other vehicles and only at really weird and specific angles
idk if that's because of where raycast originates (like, if it's too high) or if it's like only hitting the wheels, etc., but it DOES hit them sometimes

i'm thinking we just give vehicles some invisible hitbox or something similar ? it would make aiming a lot easier since players are expected to shoot and also race at the same time. but it def is doing something right now

### CODE REVIEW

File: game/src/game/components/RaycastComponent.cpp [hunk 1]
```diff
@@ -19,43 +20,53 @@ void RaycastComponent::OnInit(const ServiceProvider& service_provider)
...
+    // get the data from the raycast hit
     RaycastData raycast = physics_service_->Raycast(origin, direction).value();
+
+    // physx::PxActor* self_actor = self_shape_->getActor();
+    physx::PxActor* actor = raycast.actor;
+
+    // raycast ignores own entity
+    // if (actor == self_actor)
+    // return;
+
```
Comment (raduschirliu) [inline game/src/game/components/RaycastComponent.cpp:61]: If these aren't used anymore we might as well just delete it

File: game/src/game/components/RaycastComponent.h [hunk 2]
```diff
@@ -20,7 +21,13 @@ class RaycastComponent final : public Component,
...
+    // service dependencies
     jss::object_ptr<PhysicsService> physics_service_;
+
+    // component dependencies
+    jss::object_ptr<Transform> transform_;
+    // jss::object_ptr<BoxRigidBody> box_rigid_body_;
+
+    // the shape of the currernt entity
+    // physx::PxShape* self_shape_;
 };
```
Comment (raduschirliu) [inline game/src/game/components/RaycastComponent.h:33]: If these aren't used anymore, maybe remove entirely

**Review by raduschirliu — CHANGES_REQUESTED** [review]
Looks great! My only change requests were kinda nitpicky about removing some commented out code if we're done with it :P

**Comment by raduschirliu** [comment]
Also re: the vehicle not having a Actor or whatever, it _does_ but it's really nestled into a million different structs, it's kinda whack.

If you [look in the vehicle component here](https://github.com/raduschirliu/cpsc585/blob/main/game/src/game/components/VehicleComponent.cpp#L69) you'll see how I get access to the rigidbody and its shapes. it seems to have 5 shapes, with the first one being the cuboid thing for the body, and the other 4 being the wheels (which have colllision disabled cause it was being wonky when it was enabled)

File: game/src/game/components/RaycastComponent.cpp (hunk 1 above)
Comment (jullianpalma) [reply game/src/game/components/RaycastComponent.cpp:61]: oh good catch, i just missed them !

**Review by raduschirliu — APPROVED** [review]

**Pull request merged**